src/main/java/hudson/plugins/sidebar_link/ProjectLinks.java

```diff
@@ -83,4 +81,20 @@ public ProjectLinks newInstance(StaplerRequest req, JSONObject formData) throws
                     : null;
         }
     }
+
+    @Extension(optional = true)
+    public static class TransientActionFactoryImpl extends TransientActionFactory<Job> {
+        @Override
+        public Class<Job> type() {
+            return Job.class;
+        }
+
+        public Collection<LinkAction> createFor(Job job) {
+            ProjectLinks links = (ProjectLinks) job.getProperty(ProjectLinks.class);
```

Comment:
I meant that `job.getProperty(ProjectLinks.class)` should already return instance of `ProjectLinks` without additional casts

http://javadoc.jenkins.io/hudson/model/Job.html#getProperty-java.lang.Class-